internal/pkg/deploy/cloudformation/app.go

```diff
@@ -313,12 +373,77 @@ func (cf CloudFormation) RemoveJobFromApp(app *config.Application, jobName strin
 	return nil
 }
 
+// RemoveEnvFromAppOpts contains the parameters to call RemoveEnvFromApp.
+type RemoveEnvFromAppOpts struct {
+	App          *config.Application
+	EnvToDelete  *config.Environment
+	Environments []*config.Environment
```

Comment:
Can you think of an easier way for the Cloudformation package to get access to the environments than passing them in here? I don't want to pass a `store` into the `CloudFormation` struct if I can avoid it; that feels like too much abstraction blending to me and it would be a first for this package.

Reply:
The CFN stacks we're dealing with don't have any metadata about where the environments are deployed. Typically we fetch information about the application with a config store or deploy store object, but again, passing a store to CFN feels like inverting the level of abstraction (and may lead to a circular dependency if we use a deploystore)

Reply:
Yeah totally agreed. I'm good for leaving it as is for now and probably refactor in the following PR if we can find a good way.